txp/Pos/Binary/Txp/Core.hs

```diff
@@ -19,10 +20,18 @@ import qualified Pos.Txp.Core.Types as T
 ----------------------------------------------------------------------------
 
 instance Bi T.TxIn where
-  encode txIn = encodeListLen 2 <> encode (T.txInHash txIn) <> encode (T.txInIndex txIn)
+  -- [CSL-1526] Support the "unknown" pattern for encoding a `TxIn`, so that
+  -- we can down the line extend the `TxIn` type is a backward-compatible way.
+  encode txIn =  encodeListLen 3
+              <> encode (0 :: Word8)
+              <> encode (T.txInHash txIn)
+              <> encode (T.txInIndex txIn)
   decode = do
-    enforceSize "TxIn" 2
-    T.TxIn <$> decode <*> decode
+    enforceSize "TxIn" 3
+    tag <- decode @Word8
+    case tag of
+        0 -> T.TxIn <$> decode <*> decode
```

Comment:
Ah, I see what you mean now. Yes, I think you are right, in retrospect we should really make this resilient and always encode as a list of len 2 where the first field is the tag and the second a serialised bytestring. Leave it with me!